realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -856,13 +866,24 @@ void notifyChangeListeners() {
             //FIXME: still waiting for Core to provide a fix
             //       for crash when calling _sync_if_needed on a cleared View.
             //       https://github.com/realm/realm-core/pull/1390
-            long version = table.sync();
+            long version;
+            try {
+                version = table.sync();
+            } catch (DeleteRealmListException e) {
+                // Although this RealmResults won't be updated anymore, it is good to give user a chance to do update.
+                version = TABLE_VIEW_VERSION_REALM_LIST_DELTED;
+                RealmLog.d("The parent RealmList has been deleted already.");
+            }
```

Comment:
@cmelchior 
We have discussion with @jpsim in https://github.com/realm/realm-core/pull/1434#issuecomment-171008559 . It tends to be the most convenient way for user is just give user a empty results when this happens. @rrrlasse will make core behaviour like that in the later core release. Currently i just catch the exception and make the behaviour the same as what we could have in core in the future.

@stk1m1 Think about you might delete the Parent object when you get response from server side in another thread, and that will be super difficult for user to remove the change listener in the UI thread, with all kind of timing headaches.

BTW, logically, it is not pretend the results is empty ---- it is literally empty, because the query condition is empty (deleted).